### PR TITLE
combat-trainer.lic: Allow specifying an ambush location for better teaching

### PR DESCRIPTION
**Pull request by Sarvatt**
Changes by Majorly.

### CODE REVIEW

**Comment by rcuhljr** [comment]
Has anyone verified this has any effect? They removed location targeting teaching for TM, I can't imagine it still works for weapons, and the difficulty of attacking from stealth is against creature perception regardless of body part attacked.

**Comment by Sarvatt** [comment]
He said it was a very noticeable difference in backstab skill learned picking different body parts to ambush, the wiki agrees https://elanthipedia.play.net/Backstab_skill. I thought it was odd they removed this mechanic from everything else but ambush too but that seems to be the case.

**Comment by Sarvatt** [comment]
It's just for the amount of backstab skill in the bucket on non bipeds, not stealth.

**Comment by rcuhljr** [comment]
Oh, ambush training is just trash for backstab exp compared to khri so I didn't consider that.

File: combat-trainer.lic [hunk 1]
```diff
@@ -1721,6 +1721,9 @@ class AttackProcess
     @offhand_thrown = settings.offhand_thrown
     echo("  @offhand_thrown: #{@offhand_thrown}") if $debug_mode_ct
 
+    @ambush_location = settings.ambush_location
```
Comment (rpherbig) [inline combat-trainer.lic]: This needs an entry in `base.yaml`

I recommend defaulting it to `back`

File: combat-trainer.lic [hunk 2]
```diff
@@ -1721,6 +1721,9 @@ class AttackProcess
...
+    echo("  @amubsh_location: #{@ambush_location}") if $debug_mode_ct
```
Comment (rpherbig) [inline combat-trainer.lic]: "ambush"

File: combat-trainer.lic [hunk 3]
```diff
@@ -1784,7 +1787,11 @@ class AttackProcess
 
       if (game_state.backstab? || game_state.ambush?) && hiding?
         if (game_state.backstab? && game_state.no_stab_current_mob) || (game_state.ambush? && !game_state.backstab?)
-          command += ' back'
+          if @ambush_location
+            command += " #{@ambush_location}"
```
Comment (rpherbig) [inline combat-trainer.lic]: I think you can simplify this to `command += @ambush_location`

Comment (rpherbig) [reply combat-trainer.lic]: Though if you default `@ambush_location` to `'back'`, you can get rid of this entire conditional

Comment (Sarvatt) [reply combat-trainer.lic]: Very true, getting rid of the conditional will break people who haven't restarted dependency to get the updated base though. Should I leave the conditional in for a few days then back it out when its reasonable to expect everyone has updated, or just drop it?

Comment (rpherbig) [reply combat-trainer.lic]: I'd leave it in for a few days, with another PR to remove it

File: combat-trainer.lic (hunk 1 above)
Comment (Sarvatt) [reply combat-trainer.lic]: accounted for by https://github.com/rpherbig/dr-scripts/pull/1974/commits/c12852567a3e754a06716a664eac546b3f02350a

**Comment by Sarvatt** [comment]
Addressed everything on the branch now except the conditional and I have the conditional removal ready to go

**Pull request merged**